File: app/src/main/java/fr/jbme/raiderioapp/data/model/character/GearItem.java
package fr.jbme.raiderioapp.data.model.character;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class GearItem {

    @SerializedName("item_id")
    @Expose
    public Integer itemId;
    @SerializedName("item_level")
    @Expose
    public Integer itemLevel;
    @SerializedName("item_quality")
    @Expose
    public Integer itemQuality;
    @SerializedName("is_legion_legendary")
    @Expose
    public Boolean isLegionLegendary;
    @SerializedName("is_azerite_armor")
    @Expose
    public Boolean isAzeriteArmor;
    @SerializedName("azerite_powers")
    @Expose
    public List<AzeritePower> azeritePowers = null;
    @SerializedName("corruption")
    @Expose
    public Corruption corruption;
    @SerializedName("gems")
    @Expose
    public List<Integer> gems = null;
    @SerializedName("bonuses")
    @Expose
    public List<Integer> bonuses = null;

}

File: app/src/main/java/fr/jbme/raiderioapp/model/RIOCharacter/GearItem.kt
package fr.jbme.raiderioapp.model.RIOCharacter

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class GearItem(
    @SerializedName("item_id")
    @Expose
    var itemId: Int? = null,

    @SerializedName("item_level")
    @Expose
    var itemLevel: Int? = null,

    @SerializedName("item_quality")
    @Expose
    var itemQuality: Int? = null,

    @SerializedName("is_legion_legendary")
    @Expose
    var isLegionLegendary: Boolean? = null,

    @SerializedName("is_azerite_armor")
    @Expose
    var isAzeriteArmor: Boolean? = null,

    @SerializedName("azerite_powers")
    @Expose
    var azeritePowers: List<AzeritePower>? = null,

    @SerializedName("corruption")
    @Expose
    var corruption: Corruption? = null,

    @SerializedName("gems")
    @Expose
    var gems: List<Int>? = null,

    @SerializedName("bonuses")
    @Expose
    var bonuses: List<Int>? = null
)